var/spack/repos/builtin/packages/py-h5py/package.py

```diff
@@ -42,12 +43,13 @@ class PyH5py(PythonPackage):
     depends_on("python@3.7:", type=("build", "run"), when="@3.2:")
 
     # Build dependencies
-    depends_on("py-cython@0.23:", type="build", when="@:2")
-    depends_on("py-cython@0.29:", type=("build"), when="@3: ^python@:3.7")
-    depends_on("py-cython@0.29.14:", type=("build"), when="@3: ^python@3.8.0:3.8")
-    depends_on("py-cython@0.29.15:", type=("build"), when="@3: ^python@3.9.0:")
+    depends_on("py-cython@0.23:0", type="build", when="@:2")
+    depends_on("py-cython@0.29:0", type=("build"), when="@3: ^python@:3.7")
+    depends_on("py-cython@0.29.14:0", type=("build"), when="@3: ^python@3.8.0:3.8")
+    depends_on("py-cython@0.29.15:0", type=("build"), when="@3: ^python@3.9.0:")
     depends_on("py-pkgconfig", type="build")
     depends_on("py-setuptools", type="build")
+    # depends_on("py-setuptools@61:", type="build", when="@3.8:")
```

Comment:
Why is this commented out? This should be right.

Reply:
If this exists, then py-h5py@3.8 is not usable. I think it just makes the selection of setuptools too heavy. I have dug through packages trying to find the path the conflicts to no avail.

Here is the pipeline that builds py-h5py, it uses `py-setuptools@63`.
https://gitlab.spack.io/spack/spack/-/jobs/5843681

Reply:
It seems to concretize just fine with 61+, what do you mean it isn't usable?

Reply:
In the full `ecp-data-vis-sdk` there is a conflict somewhere. For me, that is unusable. I don't know if it is a real conflict or if the solver is just not demoting some package and failing.

Reply:
@alalazo hitting more issues with setuptools pinning. Should we split setuptools into dozens of different packages based on version, or are we getting close to allowing multiple copies of build deps?

Reply:
This conflict only affects py-h5py 3.8, right? It's still possible to concretize ecp-data-vis-sdk with py-h5py 3.7?

Reply:
No, py-h5py 3.8 is the first version to support HDF5 1.14.0 which is a new requirement for the ECP Data and Vis SDK to support VOL adapters which allow HDF5 to use the Async IO, and more importantly the DAOS IO adaptors. DAOS is the optimized IO for Aurora using Intel's Optane memory, so it will be a pretty big deal there.

My current plan, barring this py-setuptools build dep issue getting fixed, is to add some icky logic into the dependencies so I can at least get what I want in reduced SDK builds and kick the can down the road on producing VOL support in HDF5 for VisIt and py-cinemasci as a Spack core limitation. It is kind of unfortunate.

Reply:
It may be worth going through all packages in your DAG that require older setuptools and either 1) adding a new version that supports newer setuptools, or 2) reaching out to the developers and asking them if they can support newer setuptools in the next release. I'm not sure how long it will take before we can properly model multiple versions of the same package in the DAG.

Reply:
The core of the problem is VisIt, so long as it depends on VTK 8 this is going to be a problem. Everything else is either unrestricted or up to date.